geps/gep-1619.md

```diff
@@ -218,10 +474,144 @@ The following are items that we intend to resolve before we consider this GEP im
+validating attributes that have values. However, this approach may be confusing to users as uses two different API
+patterns for cookie attributes.
+
+```go
+// HttpCookie defines a cookie to achieve session persistence.
+//
+// Support: Core
+type HttpCookie struct {
+    // Name defines the cookie's name.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MaxLength=4096
+    Name String `json:"name,omitempty"`
+
+    // SameSite defines the cookie's SameSite attribute.
+    //
+    // Support: Extended
+    //
+    // +optional
+    // +kubebuilder:validation:Enum=Strict;Lax;None
+    SameSite SameSiteType `json:"sameSite,omitempty"`
+    
+    // Domain defines the cookie's Domain attribute.
+    //
+    // Support: Extended
+    //
+    // +optional
+    // +kubebuilder:validation:MaxLength=4096
+    Domain String `json:"domain,omitempty"`
+    
+    // CookieKeywords defines the cookie's attributes that have no value.
+    //
+    // Support: Extended
+    // +optional
+    CookieKeywords []CookieKeyword `json:cookieKeywords`
+}
+
+// CookieKeyword defines the cookie's attributes that have no value.
+type CookieKeyword string
+
+const (
+    // CookieKeywordsHttpOnly specifies the HttpOnly cookie attribute.
+    CookieKeywordsHttpOnly HttpOnlyMode = "HttpOnly"
+    // CookieKeywordsSecure specifies the Secure cookie attribute.
+    CookieKeywordsSecure HttpOnlyMode = "Secure"
+)
+```
+
+Taking a different approach, this GEP could design a generic approach to configuring load balancing policy. Instead of
+only having a metaresource specifically for session persistence, a metaresource called `LoadBalancerPolicy` of which
+includes a field for session persistence along with other load balancer related configuration.
+
+The `LoadBalancerPolicy` design provides tighter coupling with other load balancing configuration which help reduce CRD
+proliferation, but may limit API flexibility as it forces coupling between concepts that may not be appropriate for
+current and future implementations.
+
+```go
+// LoadBalancerPolicy provides a way to define load balancing rules
+// for a service.
+//
+// Support: Core
+type LoadBalancerPolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of LoadBalancerPolicy.
+    Spec LoadBalancerPolicySpec `json:"spec"`
+
+    // Status defines the current state of LoadBalancerPolicy.
+    Status LoadBalancerPolicyStatus `json:"status,omitempty"`
+}
+
+// LoadBalancerPolicySpec defines the desired state of
+// LoadBalancerPolicy.
+// Note: there is no Override or Default policy configuration.
+type LoadBalancerPolicySpec struct {
+    // TargetRef identifies an API object to apply policy to.
+    // Services are the only valid API target references.
+    TargetRef gatewayv1a2.PolicyTargetReference `json:"targetRef"`
+
+    // SessionPersistence defines and configures session persistence.
```

Comment:
Is this an alternative to the standalone policy above ? I think it's not a bad idea - at least for the backend-attached model.

But please add some text that "persistent session takes priority over balance algorithm" ( if it has a cookie it is used, else fallback to the balancer )

Reply:
Added a statement regarding priority.